library/std/src/io/mod.rs

```diff
@@ -2462,6 +2463,23 @@ impl<R: Read> Iterator for Bytes<R> {
             };
         }
     }
+
+    default fn size_hint(&self) -> (usize, Option<usize>) {
+        (0, None)
+    }
+}
+
+#[unstable]
+impl Iterator for Bytes<fs::File> {
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        match self.inner.metadata() {
+            Ok(metadata) => {
+                let file_length = metadata.len() as usize;
+                (0, Some(file_length))
```

Comment:
This implementation will never actually help, because in practice only the `.0` of the `size_hint()` is ever used: https://internals.rust-lang.org/t/is-size-hint-1-ever-used/8187?u=scottmcm

Reply:
That's a shame. Then this implementation would basically just add system call overhead while providing very little benefit :(